Bearded.Utilities/Geometry/CircularArc2.cs

```diff
@@ -0,0 +1,126 @@
+    {
+        public Vector2 Center { get; }
+        public float Radius { get; }
+        public Direction2 Start { get; }
+        public Direction2 End { get; }
+        public bool IsShortArc { get; }
+
+        public bool IsLongArc => !IsShortArc;
+        public Angle Angle => IsShortArc ? End - Start : oppositeAngle(End - Start);
+        public Vector2 StartPoint => Center + Radius * Start.Vector;
+        public Vector2 EndPoint => Center + Radius * End.Vector;
+
+        public float ArcLength => Angle.Radians * Radius;
+
+        /// <summary>
+        /// Returns the major arc if this arc is the minor arc, and returns the major arc if this is the minor arc.
+        /// </summary>
+        public CircularArc2 Opposite => new CircularArc2(Center, Radius, Start, End, !IsShortArc);
+
+        /// <summary>
+        /// Returns the same arc with the start and end directions swapped.
+        /// </summary>
+        public CircularArc2 Reversed => new CircularArc2(Center, Radius, End, Start, IsShortArc);
+
+        /// <summary>
+        /// Constructs the minor arc in the circle with specified center and radius between the given directions.
+        /// </summary>
+        public static CircularArc2 ShortArcBetweenDirections(
+            Vector2 center, float radius, Direction2 start, Direction2 end)
+        {
+            return new CircularArc2(center, radius, start, end, true);
+        }
+
+        /// <summary>
+        /// Constructs the major arc in the circle with specified center and radius between the given directions.
+        /// </summary>
+        public static CircularArc2 LongArcBetweenDirections(
+            Vector2 center, float radius, Direction2 start, Direction2 end)
+        {
+            return new CircularArc2(center, radius, start, end, false);
+        }
+
+        /// <summary>
+        /// Constructs the arc in the circle with specified center and radius starting in the given direction,
+        /// subtending the given angle.
+        /// </summary>
+        public static CircularArc2 FromStartAndAngle(
+            Vector2 center, float radius, Direction2 start, Angle angle)
+        {
+            if (angle.Radians < -MathConstants.TwoPi || angle.Radians >= MathConstants.TwoPi)
+            {
+                throw new ArgumentException(
+                    "Angle must be between -360° inclusive and +360° exclusive.", nameof(angle));
+            }
+
+            // When the start and end are opposite each other, the short arc goes in the negative direction (i.e. the
+            // angle is -π), so conversely -π should be considered a short arc. The long arc goes in the positive
+            // direction (i.e. the angle is -π), so conversely -π should be considered a long arc.
+            var isShortArc = angle.Radians >= -MathConstants.Pi && angle.Radians < MathConstants.Pi;
+            return new CircularArc2(center, radius, start, start + angle, isShortArc);
+        }
```

Comment:
```suggestion
        {
            var radians = angle.Radians;
            if (radians < -MathConstants.TwoPi || radians >= MathConstants.TwoPi)
            {
                throw new ArgumentException(
                    "Angle must be between -360° inclusive and +360° exclusive.", nameof(angle));
            }

            // When the start and end are opposite each other, the short arc goes in the negative direction (i.e. the
            // angle is -π), so conversely -π should be considered a short arc. The long arc goes in the positive
            // direction (i.e. the angle is -π), so conversely -π should be considered a long arc.
            var isShortArc = radians >= -MathConstants.Pi && radians < MathConstants.Pi;
            return new CircularArc2(center, radius, start, start + angle, isShortArc);
        }
```

The repetition bothered me a bit, even though I know the implementation. Completely optional, if you think this makes it less readable or something.